src/OmniSharp/Workers/Refactoring/FixUsings.cs

```diff
@@ -0,0 +1,238 @@
+
+                foreach (var node in nodes)
+                {
+                    document = workspace.GetDocument(fileName);
+                    semanticModel = await document.GetSemanticModelAsync();
+                    var sourceText = (await document.GetTextAsync());
+                    var diagnostics = semanticModel.GetDiagnostics();
+                    var location = node.Identifier.Span;
+                    var actions = new List<CodeAction>();
+                    var usingOperations = new Dictionary<string, CodeActionOperation>();
+                    var pointDiagnostics = diagnostics.Where(d => d.Location.SourceSpan.Contains(location)).ToImmutableArray();
+
+                    if (pointDiagnostics.Any())
+                    {
+                        var pointdiagfirst = pointDiagnostics.First().Location.SourceSpan;
+                        if (pointDiagnostics.Any(d => d.Location.SourceSpan != pointdiagfirst))
+                        {
+                            continue;
+                        }
+                        var context = new CodeFixContext(document, pointDiagnostics.First().Location.SourceSpan, pointDiagnostics, (a, d) => actions.Add(a), CancellationToken.None);
+                        var providers = codeActionProvider.CodeFixes;
+
+                        foreach (var provider in providers)
+                        {
+                            var register = provider.RegisterCodeFixesAsync(context);
+
+                            foreach (var action in actions)
+                            {
+                                if (action.Title.StartsWith("using"))
+                                {
+                                    var operations = await action.GetOperationsAsync(CancellationToken.None).ConfigureAwait(false);
+                                    if (operations != null)
+                                    {
+                                        foreach (var codeOperation in operations)
+                                        {
+                                            if (codeOperation != null)
+                                            {
+                                                usingOperations.Add(action.Title, codeOperation);
+                                            }
+                                        }
+                                    }
+                                }
+                            }
+                        }
+
+                        if (usingOperations.Count() == 1)
+                        {
+                            //Only one operation - apply it
+                            usingOperations.Single().Value.Apply(workspace, CancellationToken.None);
+                            updated = true;
+                        }
+                        else if (usingOperations.Count() > 1)
+                        {
+                            if (usingOperations.Keys.Any(k => k.StartsWith("using Microsoft.CodeAnalysis.CSharp")))
```

Comment:
If it's ambiguous, it's ambiguous isn't it? I don't think we should decide which using statement it should be here.